### PR TITLE
Fix russian translation and add number translation use regexp (lua pattern)

### PR DESCRIPTION
**Pull request by il-s**
Hello from Russia!

Your translation into Russian is terrible. We fixed most of the translation errors and supplemented what you have in get_text, but it is not in the translations table (in the database / additional_translation.sql file).

Your mechanism for translating numbers is very simple and, as I understand it, is based on adding and removing prefixes. However, in everyday work, very often, you need to do something more complex. The use of regular expressions (even greatly simplified as in LUA) gives much more opportunities. For example, you want to check that the number contains 11 digits and starts at 79 (RUSSIA mobile), then writing in the CUT field: "LP: ^ (79 .........) $", in the ADD field: "888% 1 ", we can add a special prefix for routing. Checking the length will save us from problems if an internal number starts at 79 or for example a local number (city numbers of 6 digits). I understand that our method is far from ideal, but perhaps it will be useful for someone and will help you make a more general solution.

### CODE REVIEW

File: freeswitch/scripts/astpp/lib/astpp.functions.lua [hunk 1]
```diff
@@ -281,6 +281,18 @@ function do_number_translation(number_translation,destination_number)
       if(tmp_str[1] == '' or tmp_str[1] == nil)then
 	return destination_number
       end
+
+      if (string.sub(tmp_str[1],0,3) == 'LP:') then
```
Comment (smrdoshi) [inline freeswitch/scripts/astpp/lib/astpp.functions.lua]: Hi @mihalevm @il-s 

Thanks for your contribution. 

We would like to understand this condition code. Can you please explain with example? 
I appreciate your efforts.

File: web_interface/astpp/application/libraries/csvreader.php [hunk 2]
```diff
@@ -103,7 +103,8 @@ function parse_lines($p_CSVLines, $config_variable, $check_header_flag = false)
 					$value = strip_slashes ( trim ( $value ) );
 					$value = preg_replace ( '#<script.*</script>#is', '', $value );
 					if (isset ( $field_key_value ) && ! empty ( $field_key_value ))
-						$content [$field_key_value] = strip_tags ( filter_var ( $value, FILTER_SANITIZE_STRING ) );
+//						$content [$field_key_value] = strip_tags ( filter_var ( $value, FILTER_SANITIZE_STRING ));
```
Comment (smrdoshi) [inline web_interface/astpp/application/libraries/csvreader.php:106]: We need this line here as some time we are getting html text in value. So we want to clean them before processing.

File: web_interface/astpp/application/views/header.php [hunk 3]
```diff
@@ -141,6 +141,7 @@ function process_alert(id,table) {
      <script src="<?php echo base_url(); ?>assets/js/jquery-1.12.4.js"></script>
      <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/bootstrap.bundle.min.js"></script>
 	 <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/bootstrap-select.js"></script>  
+     <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/i18n/defaults-<?=$_COOKIE["lang"]?>.js"></script>
```
Comment (smrdoshi) [inline web_interface/astpp/application/views/header.php:144]: Please explain the usage of this js?

**Review by smrdoshi — CHANGES_REQUESTED** [review]

File: web_interface/astpp/application/libraries/csvreader.php (hunk 2 above)
Comment (mihalevm) [reply web_interface/astpp/application/libraries/csvreader.php:106]: Some translations has html tags (for examle "Connection \<br\>Cost"). Installing laguage addon pass througth this module and filtering some strings like this. I think this bad practice, used html tags in translation, better use css-style for word wrap in tables header. But now for minimum code correction remove this filter from csvreader library.

File: web_interface/astpp/application/views/header.php (hunk 3 above)
Comment (mihalevm) [reply web_interface/astpp/application/views/header.php:144]: Project include JS libraries, such as datetimepicker, bootstrap-select. Some libraries already has translation. For examle bootstrap-select library has own localization (path in project: /web_interface/astpp/assets/js/i18n/defaults-xx_XX.js). This visual component used in some places, example (Services>Product>Edit Product>Add Destination>"Call Type, Rate Group, Country"). By using cookie variable "lang" (ru_RU) and same part of file (defaults-ru_RU.js) we can loading current translation for some JS libraries.

изображение

File: freeswitch/scripts/astpp/lib/astpp.functions.lua (hunk 1 above)
Comment (il-s) [reply freeswitch/scripts/astpp/lib/astpp.functions.lua]: Example in request text:
"For example, you want to check that the number contains 11 digits and starts at 79 (RUSSIA mobile), then writing in the CUT field: "LP: ^(79.........)$", in the ADD field: "888%1", we can add a special prefix for routing. Checking the length will save us from problems if an internal number starts at 79 or for example a local number (city numbers of 6 digits)."

The prefix "LP:" (Lua Pattern) allows you to separate the normal behavior during translation and translation using regular expressions.

**Comment by devangn** [comment]
@il-s 
Can you please share the usage example of regex base localization?

**Comment by il-s** [comment]
> Can you please share the usage example of regex base localization?

@devangn
Regex-based localization allows you to:
1. Check prefixes: "^79"
2. To remove or add prefixes: "^79(.+)$", The replacement template "%1" will remove the prefix, and the template "89%1" will change the prefix
3. Check the prefix and number length "^(79..........)$" - checks that the number starts with 79 and has a length of 11 digits

At the current moment, I completely rewrote the function of translating numbers to use ONLY regular expressions (see commit number [21e5bd118f8c934b78da5740811cd77249be5230](https://github.com/mihalevm/ASTPP/commit/21e5bd118f8c934b78da5740811cd77249be5230) ), and also select those translation that most closely meet the criterion, for example, you need to convert the number 79051234567, there are 2 rules "^7(.+)$" and "^79(. +)$". In this case, the translation function will select the translation "^79(.+)$". Because coincidence on it more.

Also, in commit [5f085be26e85134a8e96da1b04429be1797b4844](https://github.com/mihalevm/ASTPP/commit/5f085be26e85134a8e96da1b04429be1797b4844), a bug was fixed that led to the fact that during the translation of CallerID, each subsequent rule tried to translate not the initial, but already changed one.

I’m going to refine the number translation function so that it freely accepts the old syntax of the form "delete prefix \ add prefix".

**Comment by smrdoshi** [comment]
Hi @il-s 

There are SQL queries missing in your committed features, would you please share those queries?

**Pull request closed without merging**